### ISSUE TITLE
Add summary generic for dfunc class

### ISSUE THREAD

**Issue by jcarlis3** [tmcd82070/Rdistance#57]
We have `plot.dfunc` and `print.dfunc` generics for fitted detection function objects (class `dfunc`).  `summary` is another generic I think users will try.  It returns nonsense currently.  `print` actually returns something you may expect from `summary`.  @tmcd82070 , should we make a `summary.dfunc` generic that returns what `print.dfunc` currently does?  And make 'print.dfunc' return something simpler? 

```
# Example with package data
# Must be package version on GitHub (2.x), not CRAN (1.3.2)
require(Rdistance)

data("sparrowDetectionData")
data("sparrowSiteData")

x <- dfuncEstim(dist ~ shrub, likelihood="halfnorm", w.hi=100,
                detectionData=sparrowDetectionData, siteData=sparrowSiteData)
plot(x, newdata=data.frame(shrub=seq(5, 25, by=5)), nbins=5)
print(x)
summary(x)
```

**Comment by tmcd82070** [comment]
Yes.

**Comment by jcarlis3** [comment]
@mkleinsa , can you give this a crack?  Chat with me or @tmcd82070 for specifics.

**Comment by tmcd82070** [comment]
Implemented in v3.1.0 (28 aug 2023)
